src/bmi/CMakeLists.txt

```diff
@@ -1,25 +1,37 @@
-include(${PROJECT_SOURCE_DIR}/cmake/dynamic_sourced_library.cmake)
-dynamic_sourced_cxx_library(ngen_bmi "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
Dynamic sourcing is being removed because we should aim to eventually get rid of it completely. Conditional compilation is better managed by the build system rather than in the C++ files themselves, but this dynamic sourcing is preventing that from happening.